### PR TITLE
Separate SASL and server passwords

### PR DESCRIPTION
**Pull request by prawnsalad**

### CODE REVIEW

**Comment by richrd** [comment]
Nice, i assume this fixes #103 ?

File: src/commands/handlers/registration.js [hunk 1]
```diff
@@ -354,6 +356,29 @@ var handlers = {
...
+/**
+ * Only use the nick+password combo if an account has not been specifically given.
+ * If no account:{account,password} has been given, use the nick+password if available
+ */
+function getSaslAuth(handler) {
+    let options = handler.connection.options;
+    if (options.account && options.account.account) {
+        return {
+            account: options.account.account,
+            password: options.account.password || '',
```
Comment (xPaw) [inline src/commands/handlers/registration.js]: This PR looks fine, the only question I have is blank passwords. Does any auth flow support that?

File: src/commands/handlers/registration.js [hunk 2]
```diff
@@ -354,6 +356,29 @@ var handlers = {
...
+        };
+    } else if (options.account && options.password) {
```
Comment (xPaw) [inline src/commands/handlers/registration.js]: Perhaps we should return null if `options.account` object exists at all, not just passwords.

File: src/commands/handlers/registration.js [hunk 3]
```diff
@@ -354,6 +356,29 @@ var handlers = {
     }
 };
 
+/**
+ * Only use the nick+password combo if an account has not been specifically given.
+ * If no account:{account,password} has been given, use the nick+password if available
+ */
+function getSaslAuth(handler) {
+    let options = handler.connection.options;
```
Comment (xPaw) [inline src/commands/handlers/registration.js]:
```suggestion
    const options = handler.connection.options;
```

File: src/commands/handlers/registration.js [hunk 4]
```diff
@@ -131,7 +131,8 @@ var handlers = {
         ];
 
         // Optional CAPs depending on settings
-        if (handler.connection.options.password || handler.connection.options.sasl_mechanism === 'EXTERNAL') {
+        let saslAuth = getSaslAuth(handler);
```
Comment (xPaw) [inline src/commands/handlers/registration.js]:
```suggestion
        const saslAuth = getSaslAuth(handler);
```

File: src/commands/handlers/registration.js (hunk 1 above)
Comment (prawnsalad) [reply src/commands/handlers/registration.js]: I'm not sure if any support it, but there's no harm in accepting a blank password while the account name must definitely not be empty.

**Comment by prawnsalad** [comment]
@richrd if the client/application that uses this lib exposes it, yes. Otherwise it will work in the same way for backwards compatibility

**Pull request merged**